Core/src-test/com/infiniteautomation/mango/spring/service/ConfigurationTemplateServiceTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ *  Copyright (C) 2023 RadixIot LLC. All rights reserved.
+ */
+
+package com.infiniteautomation.mango.spring.service;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Test;
+
+import com.serotonin.m2m2.Common;
+import com.serotonin.m2m2.MangoTestBase;
+
+public class ConfigurationTemplateServiceTest extends MangoTestBase {
+
+    @Test
+    public void testConfigurationTemplateService() throws IOException {
+        ConfigurationTemplateService service = Common.getBean(ConfigurationTemplateService.class);
+
+        //Move template files into file store
+        Path rootPath = Common.getBean(FileStoreService.class).getPathForWrite("default", "");
+        Files.createDirectories(rootPath);
+        InputStream is = getClass().getResourceAsStream("/configTemplateTestData.csv");
+
+        Path dst = rootPath.resolve("configTemplateTestData.csv");
+        Files.copy(is, dst, StandardCopyOption.REPLACE_EXISTING);
+        List<Map<String, Object>>
+                result = service.generateConfig("default", "configTemplateTestData.csv");
+
+        assertNotNull(result);
+        assertTrue(result.size() > 0);
+    }
+    @Test
+    public void testConfigurationTemplateServiceTemplate() throws IOException {
+        ConfigurationTemplateService service = Common.getBean(ConfigurationTemplateService.class);
+
+        //Move template files into file store
+        Path rootPath = Common.getBean(FileStoreService.class).getPathForWrite("default", "");
+        Files.createDirectories(rootPath);
+        InputStream is = getClass().getResourceAsStream("/configTemplateTestData.csv");
+        Path dst = rootPath.resolve("configTemplateTestData.csv");
+        Files.copy(is, dst, StandardCopyOption.REPLACE_EXISTING);
+
+        String result = service.generateTemplate("default", "configTemplateTestData.csv","roles.mustache");
+
+        assertNotNull(result);
+        assertTrue(result.length() > 0);
+        assertTrue(result.contains("ROLE_DATAH_SiteA_1"));
+        assertTrue(result.contains("ROLE_SITE_SiteB"));
```

Comment:
This is not good enough, it needs to verify that it is valid JSON and inspect the relevant items inside the JSON. Consider using Jackson mapper to convert it to JSON nodes and then use JSON pointers to access the nodes.

Reply:
As a second step, actually import the JSON into Mango and retrieve the data points from the data point service for example.